### PR TITLE
Remove user personal details filters data

### PR DESCRIPTION
**Pull request by sahsisunny**
## Closes: #1192

API Link : https://api.realdevsquad.com/users/search?state=IDLE&state=ACTIVE&state=OOO

**Description:**
The current above API returns personal details like Phone , E-mail and GitHub Access tokens. In this PR we are removing user personal details from the response.

**Test Coverage**
image

**Screenshot**
Before
image

After
image

### CODE REVIEW

**Review by heyrandhir — DISMISSED** [review]

**Review by RitikJaiswal75 — DISMISSED** [review]

**Review by shubhamsigdar1 — APPROVED** [review]
Great work @sahsisunny

**Review by heyrandhir — APPROVED** [review]

File: controllers/users.js [hunk 1]
```diff
@@ -576,10 +576,15 @@ const filterUsers = async (req, res) => {
       return res.boom.badRequest("filter for item not provided");
     }
     const users = await userQuery.getUsersBasedOnFilter(req.query);
-
+    const sanitizedUsers = users.map((user) => {
```
Comment (heyrandhir) [inline controllers/users.js:579]: I liked `sanitizedUsers`. Sounds better than `cleanedUsers` 👍🏻

Comment (shubhamsigdar1) [reply controllers/users.js:579]: `sanitizedUsersDetail`?? :)

**Comment by prakashchoudhary07** [comment]
We should create a data access layer, which will always return the user's filtered data

**Comment by sahsisunny** [comment]
> We should create a data access layer, which will always return the user's filtered data/
> So that if passing it through that code, will remove sensitive data

Okay so can I create the issue ticket for this ?

**Pull request merged**